File: core/src/main/java/vn/loitp/restclient/RestRequestInterceptor.java
package vn.loitp.restclient;

import java.io.IOException;
import java.util.Enumeration;
import java.util.Hashtable;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class RestRequestInterceptor implements Interceptor {
    private Hashtable<String, String> headers;

    public RestRequestInterceptor() {
        headers = new Hashtable<>();
    }

    public void addHeader(String key, String value) {
        headers.put(key, value);
    }

    public void removeHeader(String key) {
        headers.remove(key);
    }

    public boolean hasHeader(String key) {
        return headers.containsKey(key);
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request.Builder builder = chain.request().newBuilder();
        if (headers != null && headers.size() > 0) {
            Enumeration<String> keys = headers.keys();
            while (keys.hasMoreElements()) {
                String key = keys.nextElement();
                String value = headers.get(key);
                builder.header(key, value);
            }
        }
        return chain.proceed(builder.build());
    }
}

File: core/src/main/java/com/restapi/restclient/RestRequestInterceptor.kt
package com.restapi.restclient

import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException
import java.util.*

class RestRequestInterceptor : Interceptor {
    private val headers = Hashtable<String, String>()

    fun addHeader(key: String, value: String) {
        headers[key] = value
    }

    fun removeHeader(key: String) {
        headers.remove(key)
    }

    fun hasHeader(key: String): Boolean {
        return headers.containsKey(key)
    }

    override fun intercept(chain: Interceptor.Chain): Response {
        val builder = chain.request().newBuilder()
        if (headers.size > 0) {
            val keys = headers.keys()

            while (keys.hasMoreElements()) {
                val key = keys.nextElement()
                val value = headers[key]

                value?.let {
                    builder.header(key, it)
                }
            }
        }
        return chain.proceed(builder.build())
    }
}
